cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl_dpc.cpp

```diff
@@ -145,11 +129,14 @@ template <typename Float, typename Bin, typename Index, typename Task>
 void train_kernel_hist_impl<Float, Bin, Index, Task>::init_params(train_context_t& ctx,
                                                                   const descriptor_t& desc,
                                                                   const table& data,
-                                                                  const table& responses) {
+                                                                  const table& responses,
+                                                                  const table& weights) {
     ctx.distr_mode_ = (comm_.get_rank_count() > 1);
 
     ctx.use_private_mem_buf_ = true;
 
+    ctx.is_weighted_ = (weights.get_row_count() == data.get_row_count());
```

Comment:
May be its better to use `weights.has_data()` and add additional check or exceptions for cases when count of rows is different